### PR TITLE
Multi Assignment Request

### PR DESCRIPTION
**Pull request by Iansberg**
This request includes assignments 4-(mailroom 2 and kata), 5 -  mailroom3, 6 unittest, and Assignment 8.

Still working on 7.

### CODE REVIEW

File: students/Sahlberg/Lesson5/mailroom3/mailroom3.py [hunk 1]
```diff
@@ -5,108 +5,154 @@
...
+                amount = donate()
+                amount_append(option, donors, amount)
+                print(email(option, amount))
+                return user_action1(dict)
+            except ValueError as err:
+                print('Something is not right.')
+                return print(err)
+
+
+def user_action2(dict):
```
Comment (natasha-aleksandrova) [inline students/Sahlberg/Lesson5/mailroom3/mailroom3.py]: use function names that represent what the work is being done:
```
def create_report
def send_thank_you
``` 
etc

File: students/Sahlberg/Lesson6/Mailroom3.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+
+#for file testing only
+if __name__ == "__main__":
+    print('this is main')
+
+else:
+
+    #for file dating
+    import datetime as dt
```
Comment (natasha-aleksandrova) [inline students/Sahlberg/Lesson6/Mailroom3.py]: imports go at the top of the file

File: students/Sahlberg/Lesson6/UnitTest.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+import unittest as u
+import Mailroom3 as m
+from Mailroom3 import amount_append
+from Mailroom3 import donate
+from Mailroom3 import print_report
+from Mailroom3 import quit
+
+
+class tests(u.TestCase):
```
Comment (natasha-aleksandrova) [inline students/Sahlberg/Lesson6/UnitTest.py]: `class MailroomTest(unittest.TestCase):`

* camel case class name
* singular names for class names
* don't use alias imports

File: students/Sahlberg/Lesson6/UnitTest.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    def test_donate(self): #Requires test input = amount below (100)
+        test_dict = {'bugs mouse': [32, 5], 'mickey bunny': {678}}
+        amount_donated = 100
+        expected = 100
+        actual = donate()
+        self.assertEqual(expected,actual)
+    """
+
+    def test_amount_append(self):
+        dict = {'Roberto' : []}
```
Comment (natasha-aleksandrova) [inline students/Sahlberg/Lesson6/UnitTest.py]: *really* bad practice - you are overwriting `dict` built-in here
notice that github is highlighting it in a different color

File: students/Sahlberg/Lesson6/UnitTest.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+    def test_email(self):
+        name = 'Bibs'
+        amount = 666.00
+        expected = 'Thank you {} for the generous donation of $ {:.2f}. We appreciate your generosity.\n\nSincerely, \n\nThe Helping R Us Team\n\n'.format(name.title(), amount)
+        actual = m.email(name, amount)
+        self.assertEqual(expected, actual)
+
+    def test_quit(self):
+        actual = quit(self)
```
Comment (natasha-aleksandrova) [inline students/Sahlberg/Lesson6/UnitTest.py]: why are you passing `self` to the mailroom function?

File: students/Sahlberg/Lesson8/Circle.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
+
+
+
+class Circle(object):
```
Comment (natasha-aleksandrova) [inline students/Sahlberg/Lesson8/Circle.py]: Lesson 8 looks good

**Comment by Iansberg** [comment]
I have too much on my plate now so after this course I will not be
completing the certificate. Right now I just want to focus on passing the
course. What do you suggest? Improve mailroom or move on and finish the
last assignments?

**Comment by natasha-aleksandrova** [comment]
In that case I would say move on to the next assignments.

**Comment by Iansberg** [comment]
Thanks. I will do that.

**Pull request merged**